File: 2025-1/sesion-de-clase-23/SoftProg/SoftProgDBManager/src/main/java/pe/edu/pucp/inf30/softprog/config/MySQLDBManager.java
package pe.edu.pucp.inf30.softprog.config;

import java.io.IOException;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

/**
 *
 * @author eric
 */
public class MySQLDBManager extends DBManager {
    protected MySQLDBManager() throws IOException {
        super();
    }
    
    @Override
    public Connection getConnection() throws SQLException, ClassNotFoundException {
        try {
            /* 
            Por ahora creamos una conexion cada vez que se necesita acceder a la base de datos, 
            por ser una aplicacion academica es una practica aceptable, en un sistema productivo
            se debe usar un pool de conexiones.
            */
            Class.forName("com.mysql.cj.jdbc.Driver");
            String cadenaConexion = cadenaConexion(host, puerto, esquema);
            return DriverManager.getConnection(cadenaConexion, usuario, password);
        }
        catch (ClassNotFoundException | SQLException e) {
            System.err.println(e);
            throw e;
        }
    }
    
    @Override
    protected String cadenaConexion(String host, int puerto, String esquema) {
        return String.format("jdbc:mysql://%s:%d/%s?useSSL=false&allowPublicKeyRetrieval=true", host, puerto, esquema);
    }
}

File: 2025-2/sesion-de-clase-23/dotnet/SoftProgDBManager/MySQLDBManager.cs
using System;
using System.Data.Common;
using MySql.Data.MySqlClient;
using PUCP.SoftProg.Db.Utils;

namespace PUCP.SoftProg.Db {
    public class MySQLDBManager : DBManager {
        private static MySQLDBManager instancia;

        protected MySQLDBManager() { }

        protected MySQLDBManager(string host, int puerto, string esquema, 
                                 string usuario, string password)
            : base(host, puerto, esquema, usuario, password, TipoDB.MySQL) {
        }

        public static MySQLDBManager GetInstance(string host, 
                                                 int puerto, 
                                                 string esquema, 
                                                 string usuario, 
                                                 string password) {
            if (instancia == null) {
                instancia = new MySQLDBManager(host, puerto, esquema, usuario, password);
            }
            return instancia;
        }

        public override DbConnection GetConnection() {
            try {
                /*
                Por ahora creamos una conexión cada vez que se necesita acceder
                a la base de datos. Para una aplicación académica es aceptable,
                pero en un sistema productivo se debe usar un pool de conexiones.
                */
                string cadenaConexion = CadenaConexion();
                return new MySqlConnection(cadenaConexion);
            }
            catch (Exception e) {
                Console.Error.WriteLine(e);
                throw;
            }
        }
    }
}
